WRFv4.5 not compatible with intel llvm compilers

**Describe the bug**
In the later half of 2023, intel will be mothballing their intel compilers (icc,ifort,icpc) for new intel compilers that are built on the llvm architecture.

These new commands are:

> export CC=icx
export CXX=icpx
export FC=ifx
export F77=ifx
export F90=ifx
export MPIFC="mpiifort -fc=ifx"
export MPIF77="mpiifort -fc=ifx"
export MPIF90="mpiifort -fc=ifx"
export MPICC="mpiicc -cc=icx"
export MPICXX="mpiicpc -cxx=icpx"

wrf v4.5 is not compatibile with these new llvm compilers and will make the intel compilers for new users useless.  Anyone who has intel compilers and update their intel compiler packages will also lose the ability to use them.  This is a MAJOR problem for the WRF community.


**To Reproduce**
Steps to reproduce the behavior:

Attached is a couple of scripts that show how to install the required libraries needed to run WRF using the intel llvm compilers.  
All system tests pass so the new intel llvm compilers 

However, when you change the old intel compilers listed in the configure.wrf for the new llvm compilers the wrf will not install.



**Expected behavior**
wrf v4.5 expected to install with new llvm installers without having to modify any major files.



**Attachments**
wrf_4.5_new_intel_commands.sh:  This is a script that installs all the required libraries for WRF.  It works when using the old intel compilers as it is currently being used on github.
wrf.new.inte.compile.log: Build log using the new intel compilers.
configure.wrf: this is the configure file built using the new intel llvm compilers.



~~WRF_v4.5_Intel_LLVM_Compilers.tar.gz~~
*Edit: link removed*

In your `wrf.compile.log`, one of the first errors that I see suggests a problem with your MPI compiler wrappers not accepting `-fc=...`:
```
make[3]: Entering directory '/home/workhorse/WRF_Intel/WRFV4.5/external/io_pnetcdf'
make[3]: c=ifx -real-size 32 -i4  -O3 -ip -fp-model precise -w -ftz -align all -fno-alias -FR -convert big_endian  : No such file or directory
make[3]: *** No rule to make target 'c=ifx -real-size 32 -i4  -O3 -ip -fp-model precise -w -ftz -align all -fno-alias -FR -convert big_endian  '.  Stop.
make[3]: Leaving directory '/home/workhorse/WRF_Intel/WRFV4.5/external/io_pnetcdf'
make[2]: [../configure.wrf:313: wrfio_pnf] Error 2 (ignored)
```
For what it's worth, initial testing on our end with the Intel oneAPI compilers hasn't turned up any problems in the WRF source code.

I'm closing this issue, but if you can point to a specific problem in the WRF code or WRF configure/compile/configure.defaults, feel free to re-open it an add more details.

> In your `wrf.compile.log`, one of the first errors that I see suggests a problem with your MPI compiler wrappers not accepting `-fc=...`:
> 
> ```
> make[3]: Entering directory '/home/workhorse/WRF_Intel/WRFV4.5/external/io_pnetcdf'
> make[3]: c=ifx -real-size 32 -i4  -O3 -ip -fp-model precise -w -ftz -align all -fno-alias -FR -convert big_endian  : No such file or directory
> make[3]: *** No rule to make target 'c=ifx -real-size 32 -i4  -O3 -ip -fp-model precise -w -ftz -align all -fno-alias -FR -convert big_endian  '.  Stop.
> make[3]: Leaving directory '/home/workhorse/WRF_Intel/WRFV4.5/external/io_pnetcdf'
> make[2]: [../configure.wrf:313: wrfio_pnf] Error 2 (ignored)
> ```
> 
> For what it's worth, initial testing on our end with the Intel oneAPI compilers hasn't turned up any problems in the WRF source code.
> 
> I'm closing this issue, but if you can point to a specific problem in the WRF code or WRF configure/compile/configure.defaults, feel free to re-open it an add more details.

Were you using the new llvm intel compilers?  I'll try to see what's going on on my side.

> > In your `wrf.compile.log`, one of the first errors that I see suggests a problem with your MPI compiler wrappers not accepting `-fc=...`:
> > ```
> > make[3]: Entering directory '/home/workhorse/WRF_Intel/WRFV4.5/external/io_pnetcdf'
> > make[3]: c=ifx -real-size 32 -i4  -O3 -ip -fp-model precise -w -ftz -align all -fno-alias -FR -convert big_endian  : No such file or directory
> > make[3]: *** No rule to make target 'c=ifx -real-size 32 -i4  -O3 -ip -fp-model precise -w -ftz -align all -fno-alias -FR -convert big_endian  '.  Stop.
> > make[3]: Leaving directory '/home/workhorse/WRF_Intel/WRFV4.5/external/io_pnetcdf'
> > make[2]: [../configure.wrf:313: wrfio_pnf] Error 2 (ignored)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > For what it's worth, initial testing on our end with the Intel oneAPI compilers hasn't turned up any problems in the WRF source code.
> > I'm closing this issue, but if you can point to a specific problem in the WRF code or WRF configure/compile/configure.defaults, feel free to re-open it an add more details.
> 
> Were you using the new llvm intel compilers? I'll try to see what's going on on my side.

Was able to get the the libraries to install properly but the WRF doesn't recognize the llvm compilers.  I'm not sure what to do here.

--- a/arch/configure.defaults
+++ b/arch/configure.defaults
@@ -2125,6 +2125,52 @@ NETCDFPAR_BUILD =       CONFIGURE_NETCDFPAR_BUILD
 #insert new stanza here
 
 ###########################################################
+#ARCH    Linux x86_64 ppc64le i486 i586 i686, Intel oneAPI compiler with icx #serial smpar dmpar dm+sm
+# This mirrors the above Intel build but changes ifort=>ifx icc=>icx
+
+DESCRIPTION     =       INTEL ($SFC/$SCC) : oneAPI LLVM
+DMPARALLEL      =       # 1
+OMPCPP          =       # -D_OPENMP
+OMP             =       # -qopenmp -fpp -auto
+OMPCC           =       # -qopenmp -auto
+SFC             =       ifx
+SCC             =       icx
+CCOMP           =       icx
+DM_FC           =       mpif90 -f90=$(SFC)
+DM_CC           =       mpicc -cc=$(SCC)
+FC              =       CONFIGURE_FC
+CC              =       CONFIGURE_CC
+LD              =       $(FC)
+RWORDSIZE       =       CONFIGURE_RWORDSIZE
+PROMOTION       =       -real-size `expr 8 \* $(RWORDSIZE)` -i4
+ARCH_LOCAL      =       -DNONSTANDARD_SYSTEM_FUNC  CONFIGURE_D_CTSM
+CFLAGS_LOCAL    =       -w -O3 -ip #-xHost -fp-model fast=2 -no-prec-div -no-prec-sqrt -ftz -no-multibyte-chars # -DRSL0_ONLY
+LDFLAGS_LOCAL   =       -ip #-xHost -fp-model fast=2 -no-prec-div -no-prec-sqrt -ftz -align all -fno-alias -fno-common
+CPLUSPLUSLIB    =
+ESMF_LDFLAG     =       $(CPLUSPLUSLIB)
+FCOPTIM         =       -O3
+FCREDUCEDOPT	=       $(FCOPTIM)
+FCNOOPT		=       -O0 -fno-inline -no-ip
+FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_created,bounds,format,output_conversion,pointers,uninit -ftrapuv -unroll0 -u
+FORMAT_FIXED    =       -FI
+FORMAT_FREE     =       -FR
+FCSUFFIX        =
+BYTESWAPIO      =       -convert big_endian
+RECORDLENGTH    =       -assume byterecl
+FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common
+FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
+MODULE_SRCH_FLAG =
+TRADFLAG        =      CONFIGURE_TRADFLAG
+CPP             =      /lib/cpp CONFIGURE_CPPFLAGS
+AR              =      ar
+ARFLAGS         =      ru
+M4              =      m4
+RANLIB          =      ranlib
+RLFLAGS		=
+CC_TOOLS        =      $(SCC)
+NETCDFPAR_BUILD	=      CONFIGURE_NETCDFPAR_BUILD
+
+###########################################################
 #ARCH    Linux aarch64, GCC compiler OpenMPI # serial smpar dmpar dm+sm
 #
 DESCRIPTION     =      GCC ($SFC/$SCC): Aarch64

